go/upgrade/api/api.go

```diff
@@ -100,21 +115,35 @@ func (d *Descriptor) Equals(other *Descriptor) bool {
 
 // ValidateBasic does basic validation checks of the upgrade descriptor.
 func (d Descriptor) ValidateBasic() error {
-	if d.V < minDescriptorVersion || d.V > maxDescriptorVersion {
-		return fmt.Errorf("invalid upgrade descriptor version (min: %d max: %d)",
-			minDescriptorVersion,
-			maxDescriptorVersion,
+	if d.V < MinDescriptorVersion || d.V > MaxDescriptorVersion {
+		return fmt.Errorf("invalid upgrade descriptor version: %d (min: %d max: %d)",
+			d.V,
+			MinDescriptorVersion,
+			MaxDescriptorVersion,
 		)
 	}
-	if d.Handler == "" {
-		return fmt.Errorf("empty descriptor handler")
+	if len(d.Handler) < MinUpgradeHandlerLength || len(d.Handler) > MaxUpgradeHandlerLength {
+		return fmt.Errorf("invalid upgrade descriptor handler length: %d (min: %d max: %d)",
+			len(d.Handler),
+			MinUpgradeHandlerLength,
+			MaxUpgradeHandlerLength,
+		)
 	}
-	empty := version.ProtocolVersions{}
-	if d.Target == empty {
-		return fmt.Errorf("empty target version")
+	if d.Target.ConsensusProtocol == InvalidUpgradeTargetSubversion {
```

Comment:
These checks should probably be moved into `version` package. Maybe a `Validate`/`ValidateBasic` method on the `Version` and `ProtocolVersion` structs.

Reply:
Agreed. Done.